// @urbint/silica/tailwind-preset-v2.js
/* eslint-disable @typescript-eslint/no-var-requires */
const defaultTheme = require('tailwindcss/defaultTheme');

module.exports = {
  theme: {
    extend: {
      colors: {
        neutrals: {
          primary: {
            DEFAULT: 'rgba(4, 30, 37, 1)',
            light: 'rgba(255, 255, 255, 1)',
          },
          secondary: {
            DEFAULT: 'rgba(60, 79, 85, 1)',
            light: 'rgba(192, 199, 201, 1)',
          },
          tertiary: 'rgba(131, 146, 150, 1)',
          disabled: 'rgba(164, 171, 174, 0.38)',
        },
        backgrounds: {
          white: 'rgba(255, 255, 255, 1)',
          gray: 'rgba(248, 248, 248, 1)',
          overlay: 'rgba(164, 171, 174, 0.38)',
        },
        actions: {
          primary: {
            DEFAULT: 'rgba(0, 160, 204, 1)',
            hover: 'rgba(74, 188, 220, 1)',
            active: 'rgba(0, 88, 112, 1)',
            disabled: 'rgba(153, 201, 214, 1)',
          },
          secondary: {
            DEFAULT: 'rgba(255, 255, 255, 1)',
            hover: 'rgba(248, 248, 248, 1)',
            active: 'rgba(233, 233, 233, 1)',
            disabled: 'rgba(236, 236, 236, 1)',
          },
          ghost: {
            DEFAULT: 'transparent',
            hover: 'rgba(4, 30, 37, 0.03)',
            active: 'rgba(4, 30, 37, 0.07)',
            disabled: 'transparent',
          },
          danger: {
            DEFAULT: 'rgba(231, 24, 62, 1)',
            hover: 'rgba(235, 61, 93, 1)',
            active: 'rgba(215, 24, 60, 1)',
            disabled: 'rgba(230, 151, 166, 1)',
          },
        },
        feedback: {
          info: {
            content: 'rgba(60, 135, 221, 1)',
            background: 'rgba(242, 247, 253, 1)',
            border: 'rgba(103, 162, 228, 1)',
          },
          success: {
            content: 'rgba(43, 163, 41, 1)',
            background: 'rgba(244, 251, 244, 1)',
            border: 'rgba(116, 202, 114, 1)',
          },
          warning: {
            content: 'rgba(214, 150, 0, 1)',
            background: 'rgba(255, 248, 229, 1)',
            border: 'rgba(232, 186, 48, 1)',
          },
          error: {
            content: 'rgba(231, 24, 62, 1)',
            background: 'rgba(254, 246, 247, 1)',
            border: 'rgba(237, 80, 108, 1)',
          },
        },
        brand: {
          navy: 'rgba(0, 63, 83, 1)',
          blue: 'rgba(0, 160, 204, 1)',
        },
        borders: {
          DEFAULT: 'rgba(189, 193, 195, 1)',
          disabled: 'rgba(233, 233, 233, 1)',
          hover: 'rgba(171, 178, 180, 1)',
          active: 'rgba(0, 160, 204, 1)',
          danger: 'rgba(231, 24, 62, 1)',
        },
        badges: {
          gray: 'rgba(248, 248, 248, 1)',
          red: 'rgba(207, 61, 61, 1)',
          orange: 'rgba(183, 94, 11, 1)',
          blue: 'rgba(62, 112, 212, 1)',
          green: {
            DEFAULT: 'rgba(35, 137, 20, 1)',
            neon: 'rgba(8, 133, 116, 1)',
          },
          darkgray: 'rgba(60, 79, 85, 1)',
          purple: 'rgba(151, 71, 255, 1)',
        },
      },
      fontFamily: {
        sans: ['Inter', ...defaultTheme.fontFamily.sans],
      },
      fontSize: {
        'page-heading': ['1.625rem', '1.5'],
        'section-heading': ['1.375rem', '1.5'],
        'section-subheading': ['1.25rem', '1.5'],
        subheading: ['1.125rem', '1.5'],
        'action-label': ['1rem', '1.35'],
        'body-text': ['1rem', '1.3'],
        'component-label': ['0.875rem', '1.5'],
        'caption-text': ['0.813rem', '1.5'],
      },
      fontWeight: {
        'page-heading': 400,
        'section-heading': 600,
        'section-subheading': 600,
        subheading: 600,
        'action-label': 600,
        'body-text': 400,
        'component-label': 600,
        'caption-text': 400,
      },
      boxShadow: {
        'elevation-inset-10': 'inset 0px 1px 4px rgba(0, 0, 0, 0.18)',
        'elevation-inset-5': 'inset 0px 0.5px 2px rgba(0, 0, 0, 0.09)',
        'elevation-5': '0px 0.5px 2px rgba(0, 0, 0, 0.09)',
        'elevation-10': '0px 1px 3px rgba(0, 0, 0, 0.18)',
        'elevation-20': '0px 2px 8px rgba(0, 0, 0, 0.18)',
        'elevation-30': '0px 3px 12px 1px rgba(0, 0, 0, 0.18)',
        'elevation-40': '0px 8px 24px 4px rgba(0, 0, 0, 0.18)',
      },
    },
  },
};
