// @urbint/silica/tailwind-preset.js
module.exports = {
  theme: {
    extend: {
      colors: {
        brand: {
          urbint: {
            10: '#F0FBFE',
            20: '#C8EEF9',
            30: '#61B9D1',
            40: '#00A0CC',
            50: '#007899',
            60: '#003F53',
          },
          gray: {
            10: '#F7F8F8',
            20: '#EDEFF0',
            30: '#D2D7D8',
            40: '#BEC5C6',
            50: '#A0AAAC',
            60: '#6E7D81',
            70: '#43565B',
            80: '#041E25',
          },
        },
        data: {
          red: {
            10: '#F5D1D1',
            20: '#E78D8D',
            30: '#D44242',
            40: '#A82424',
            50: '#541212',
            60: '#320B0B',
          },
          orange: {
            10: '#F3DDC9',
            20: '#E8AB73',
            30: '#B75E0B',
            40: '#A55F1D',
            50: '#753E0A',
            60: '#4B2707',
          },
          yellow: {
            10: '#F5E8BD',
            20: '#EED272',
            30: '#EEBF13',
            40: '#B18E10',
            50: '#7C640E',
            60: '#493B08',
          },
          teal: {
            10: '#CDEEEB',
            20: '#94DBD3',
            30: '#088574',
            40: '#1C7D72',
            50: '#044C52',
            60: '#032D30',
          },
          blue: {
            10: '#DDE8FD',
            20: '#97B4ED',
            30: '#3E70D4',
            40: '#1F4EAD',
            50: '#153779',
            60: '#0F2652',
          },
          purple: {
            10: '#E4E0FF',
            20: '#C4BCF5',
            30: '#705FE3',
            40: '#4F3EBB',
            50: '#2B1F7A',
            60: '#1D1452',
          },
          green: {
            10: '#CFECCA',
            20: '#91D586',
            30: '#238914',
            40: '#3A842E',
            50: '#295E21',
            60: '#173513',
          },
          fuchsia: {
            10: '#F9E7F9',
            20: '#EEB9EF',
            30: '#B24FB4',
            40: '#A327A5',
            50: '#661867',
            60: '#3D0E3E',
          },
        },
        system: {
          info: {
            10: '#F2F7FD',
            20: '#BED7F4',
            30: '#67A2E4',
            40: '#3C87DD',
            50: '#226DC3',
            60: '#133D6C',
          },
          success: {
            10: '#F4FBF4',
            20: '#C6ECC6',
            30: '#74CA72',
            40: '#2BA329',
            50: '#227E20',
            60: '#114110',
          },
          warning: {
            10: '#FFF8E5',
            20: '#FFE7B8',
            30: '#E8BA30',
            40: '#D69600',
            50: '#BD7800',
            60: '#754400',
          },
          error: {
            10: '#FEF6F7',
            20: '#F8BAC5',
            30: '#ED506C',
            40: '#E7183E',
            50: '#AF122F',
            60: '#730C1F',
          },
        },
        neutral: {
          shade: {
            100: '#041E25',
            75: 'rgba(4, 30, 37, 0.75)',
            58: 'rgba(4, 30, 37, 0.58)',
            38: 'rgba(4, 30, 37, 0.38)',
            26: 'rgba(4, 30, 37, 0.26)',
            18: 'rgba(4, 30, 37, 0.18)',
            7: 'rgba(4, 30, 37, 0.07)',
            3: 'rgba(4, 30, 37, 0.03)',
          },
          light: {
            100: '#FFFFFF',
            77: 'rgba(255, 255, 255, 0.77)',
            60: 'rgba(255, 255, 255, 0.60)',
            43: 'rgba(255, 255, 255, 0.43)',
            35: 'rgba(255, 255, 255, 0.35)',
            25: 'rgba(255, 255, 255, 0.25)',
            16: 'rgba(255, 255, 255, 0.16)',
            8: 'rgba(255, 255, 255, 0.08)',
          },
        },
      },
      boxShadow: {
        30: '0px 3px 12px 1px rgba(4, 30, 37, 0.18)',
        20: '0px 2px 8px rgba(4, 30, 37, 0.18);',
        10: '0px 1px 3px rgba(4, 30, 37, 0.18)',
        5: '0px 0.5px 2px rgba(4, 30, 37, 0.18)',
      },
      fontSize: {
        xs: ['.6875rem', '1.25'],
        tiny: ['.8125rem', '1.25'],
        sm: ['.875rem', '1.25'],
        base: ['1rem', '1.25'],
        lg: ['1.125rem', '1.5'],
        xl: ['1.375rem', '1.5'],
        '2xl': ['1.625rem', '1.5'],
        '3xl': ['2rem', '1.5'],
        '4xl': ['2.375rem', '1.5'],
        '5xl': ['3rem', '1.5'],
      },
    },
  },
  variants: {
    extend: {
      transform: ['group-hover'],
    },
  },
};
