// @urbint/silica/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var tslib = require('tslib');
var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var reactDom = require('react-dom');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n["default"] = e;
	return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);

var classnames = {exports: {}};

/*!
  Copyright (c) 2018 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/

(function (module) {
/* global define */

(function () {

	var hasOwn = {}.hasOwnProperty;

	function classNames() {
		var classes = [];

		for (var i = 0; i < arguments.length; i++) {
			var arg = arguments[i];
			if (!arg) continue;

			var argType = typeof arg;

			if (argType === 'string' || argType === 'number') {
				classes.push(arg);
			} else if (Array.isArray(arg)) {
				if (arg.length) {
					var inner = classNames.apply(null, arg);
					if (inner) {
						classes.push(inner);
					}
				}
			} else if (argType === 'object') {
				if (arg.toString === Object.prototype.toString) {
					for (var key in arg) {
						if (hasOwn.call(arg, key) && arg[key]) {
							classes.push(key);
						}
					}
				} else {
					classes.push(arg.toString());
				}
			}
		}

		return classes.join(' ');
	}

	if (module.exports) {
		classNames.default = classNames;
		module.exports = classNames;
	} else {
		window.classNames = classNames;
	}
}());
}(classnames));

var cx = classnames.exports;

var _path$u;

function _extends$w() { _extends$w = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$w.apply(this, arguments); }

function SvgArrowheadOutline(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$w({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$u || (_path$u = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M11.69 15.05a.242.242 0 01.189-.113l7.892-.491c.228-.014.313-.316.128-.456L5.387 3.05c-.17-.129-.408.013-.386.23L6.89 21.774c.024.235.32.31.445.113l4.356-6.837zm-3.436.92c.024.236.32.31.445.113l1.903-2.987a.242.242 0 01.189-.112l3.447-.215c.228-.014.313-.316.128-.456L7.783 7.35c-.17-.128-.407.014-.385.23l.856 8.39z"
  })));
}

var _path$t;

function _extends$v() { _extends$v = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$v.apply(this, arguments); }

function SvgAsterisk(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$v({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$t || (_path$t = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M10.875 9.28V3.5a.5.5 0 01.5-.5h1.25a.5.5 0 01.5.5v5.788l4.094-4.094a.5.5 0 01.707 0l.884.884a.5.5 0 010 .707l-4.09 4.09h5.78a.5.5 0 01.5.5v1.25a.5.5 0 01-.5.5h-5.78l4.09 4.09a.5.5 0 010 .707l-.884.884a.5.5 0 01-.707 0l-4.094-4.094V20.5a.5.5 0 01-.5.5h-1.25a.5.5 0 01-.5-.5v-5.78l-4.086 4.086a.5.5 0 01-.707 0l-.884-.884a.5.5 0 010-.707l4.09-4.09H3.5a.5.5 0 01-.5-.5v-1.25a.5.5 0 01.5-.5h5.788l-4.09-4.09a.5.5 0 010-.707l.884-.884a.5.5 0 01.707 0l4.086 4.086z",
    fill: "currentColor"
  })));
}

var _path$s;

function _extends$u() { _extends$u = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$u.apply(this, arguments); }

function SvgAttachment(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$u({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$s || (_path$s = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M18.538 6.914a3.195 3.195 0 00-1.516-1.491 3.259 3.259 0 00-3.666.618l-7.212 7.097a3.308 3.308 0 00-.512 4.09l.025.042c.224.366.521.683.874.931l.353.248c.68.478 1.565.572 2.333.248.39-.165.73-.428.982-.765a.147.147 0 01.204-.03l1.492 1.085a.142.142 0 01.03.201 4.53 4.53 0 01-1.862 1.45 4.602 4.602 0 01-4.427-.47l-.353-.248a5.258 5.258 0 01-1.466-1.563l-.026-.042a5.395 5.395 0 01.834-6.671l7.213-7.097a5.432 5.432 0 016.11-1.03 5.325 5.325 0 012.527 2.486 5.219 5.219 0 01-1.047 6.013l-4.17 4.104a3.257 3.257 0 01-3.225.788 3.195 3.195 0 01-2.13-2.097 3.133 3.133 0 01.8-3.172l1.944-1.912.4-.4a2.62 2.62 0 013.56-.136c.083.072.083.2.005.277l-1.224 1.224c-.078.078-.204.072-.307.032a.49.49 0 00-.528.109l-.4.4-1.932 1.9a1.044 1.044 0 00-.267 1.057c.105.334.37.596.71.699.38.115.794.014 1.075-.263l4.17-4.104a3.131 3.131 0 00.629-3.608z"
  })));
}

var _path$r, _path2$f;

function _extends$t() { _extends$t = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$t.apply(this, arguments); }

function SvgCalculator(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$t({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$r || (_path$r = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M4 5a3 3 0 013-3h10a3 3 0 013 3v14a3 3 0 01-3 3H7a3 3 0 01-3-3V5zm2 0a1 1 0 011-1h10a1 1 0 011 1v3H6V5zm12 5v9a1 1 0 01-1 1H7a1 1 0 01-1-1v-9h12z"
  })), _path2$f || (_path2$f = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M9.5 12a1 1 0 11-2 0 1 1 0 012 0zM9.5 15a1 1 0 11-2 0 1 1 0 012 0zM9.5 18a1 1 0 11-2 0 1 1 0 012 0zM13 12a1 1 0 11-2 0 1 1 0 012 0zM13 15a1 1 0 11-2 0 1 1 0 012 0zM13 18a1 1 0 11-2 0 1 1 0 012 0zM16.5 12a1 1 0 11-2 0 1 1 0 012 0zM16.5 15a1 1 0 11-2 0 1 1 0 012 0zM16.5 18a1 1 0 11-2 0 1 1 0 012 0z"
  })));
}

var _path$q;

function _extends$s() { _extends$s = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$s.apply(this, arguments); }

function SvgCamera(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$s({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$q || (_path$q = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M8.06 5.457A3 3 0 0110.632 4h2.736a3 3 0 012.572 1.457l.335.558a1 1 0 00.857.485H19a3 3 0 013 3V18a3 3 0 01-3 3H5a3 3 0 01-3-3V9.5a3 3 0 013-3h1.868a1 1 0 00.857-.486l.335-.557zM10.632 6a1 1 0 00-.857.486l-.335.557A3 3 0 016.868 8.5H5a1 1 0 00-1 1V18a1 1 0 001 1h14a1 1 0 001-1V9.5a1 1 0 00-1-1h-1.868a3 3 0 01-2.572-1.457l-.335-.558A1 1 0 0013.368 6h-2.736zM12 11.5a2 2 0 100 4 2 2 0 000-4zm-4 2a4 4 0 118 0 4 4 0 01-8 0z"
  })));
}

var _circle$1;

function _extends$r() { _extends$r = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$r.apply(this, arguments); }

function SvgCircle(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$r({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _circle$1 || (_circle$1 = /*#__PURE__*/React__namespace.createElement("circle", {
    cx: 12,
    cy: 12,
    r: 10,
    fill: "currentColor"
  })));
}

var _path$p, _path2$e;

function _extends$q() { _extends$q = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$q.apply(this, arguments); }

function SvgCrossHammerWrench(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$q({
    xmlns: "http://www.w3.org/2000/svg",
    width: "1em",
    height: "1em",
    viewBox: "0 0 47 39",
    fill: "none"
  }, props), _path$p || (_path$p = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M43.767 14.954a8.54 8.54 0 002.321-4.191 8.538 8.538 0 00-.3-4.78l-.392-1.089-6.158 6.158-3.714-3.71 6.148-6.149L40.6.796a8.581 8.581 0 00-10.434 3.932 8.584 8.584 0 00-.888 5.876l-5.865 5.854-2.053-1.973a9.764 9.764 0 01-1.628-2.363 11.741 11.741 0 01.477-3.426 5.31 5.31 0 011.607-2.8 5.309 5.309 0 012.931-1.352 1.1 1.1 0 00.984-1.09l.01-2.346a1.097 1.097 0 00-.332-.785 1.174 1.174 0 00-.8-.323c-7.067.097-10.465 2.76-10.665 2.932L7.218 9.655l-.091 1.633a1.305 1.305 0 01-1.257 1.236l-1.583.062-3.219 3.291a1.319 1.319 0 00.014 1.836l5.794 5.796a1.32 1.32 0 00.916.377 1.319 1.319 0 00.92-.365l3.295-3.211.07-1.583a1.303 1.303 0 011.23-1.253l1.16-.069a9.57 9.57 0 012.33 1.613l1.984 2.069-9.838 9.823a4.692 4.692 0 00-1.384 3.333 4.532 4.532 0 001.293 3.225l.12.154A4.72 4.72 0 0012.305 39a4.72 4.72 0 003.333-1.378l9.691-9.704 9.53 9.935a3.746 3.746 0 002.703 1.146c.227 0 .454-.02.677-.06a3.82 3.82 0 002.841-2.3 3.745 3.745 0 00.18-2.192 3.743 3.743 0 00-1.083-1.914l-9.932-9.53 5.718-5.718a8.593 8.593 0 007.803-2.331zM14.604 36.579a3.25 3.25 0 01-2.227.962 3.25 3.25 0 01-2.273-.85l-.12-.154a3.25 3.25 0 01-.95-2.296c0-.86.342-1.687.95-2.296l9.824-9.802 4.517 4.71-9.72 9.726zm25.129-.53a2.372 2.372 0 01-1.76 1.44 2.287 2.287 0 01-2.06-.657L17.85 17.99a11.276 11.276 0 00-2.932-1.995l-.166-.077-1.53.093a2.77 2.77 0 00-1.818.823 2.77 2.77 0 00-.792 1.83l-.043 1.003-2.756 2.694-5.572-5.572 2.694-2.764.998-.037a2.766 2.766 0 002.657-2.623l.058-1.07 6.27-6.273c.03-.025 3.079-2.373 9.352-2.544V3.129a6.83 6.83 0 00-3.499 1.749 6.83 6.83 0 00-1.965 3.383 13.376 13.376 0 00-.546 4.026l.066.303a11.352 11.352 0 002.002 2.95l18.86 18.068a2.244 2.244 0 01.543 2.44h.002zm-4.226-20.39l-6.326 6.332-4.71-4.515 6.43-6.417-.106-.407a7.135 7.135 0 011.862-6.816 7.06 7.06 0 016.283-1.979l-4.614 4.61a1.236 1.236 0 00-.362.874c0 .327.13.642.362.873l4.036 4.037a1.266 1.266 0 001.748 0l4.607-4.607a7.117 7.117 0 01-8.797 8.123l-.413-.108z",
    fill: "currentColor"
  })), _path2$e || (_path2$e = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M14.753 31.805a2.545 2.545 0 00-1.8-.746 2.546 2.546 0 00-1.801 4.347 2.545 2.545 0 001.8.747c.676 0 1.324-.268 1.801-.746a2.551 2.551 0 00.745-1.801 2.55 2.55 0 00-.745-1.801zm-1.036 2.566a1.104 1.104 0 01-1.53 0 1.082 1.082 0 111.53 0z",
    fill: "currentColor"
  })));
}

var _path$o, _path2$d;

function _extends$p() { _extends$p = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$p.apply(this, arguments); }

function SvgCrossLightning(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$p({
    width: "1em",
    height: "1em",
    viewBox: "0 0 25 31",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$o || (_path$o = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M19.4 16.955l2.34-2.805c.45-.54.105-1.335-.6-1.41l-7.065-.69c-.105 0-.18-.105-.15-.21L16.37.71c.045-.18-.195-.285-.315-.135l-7.725 9.99 11.07 6.39zm-13.02-3.87L4.115 16.01c-.42.54-.075 1.305.615 1.38l6.255.675c.105 0 .18.12.15.21l-3.3 11.985c-.045.18.195.3.315.15l9.18-11.01-10.965-6.33.015.015z",
    fill: "currentColor"
  })), _path2$d || (_path2$d = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M24.624 21.941L1.15 8.39.408 9.675l23.474 13.552.742-1.286z",
    fill: "currentColor"
  })));
}

var _path$n, _path2$c;

function _extends$o() { _extends$o = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$o.apply(this, arguments); }

function SvgExcavator(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$o({
    xmlns: "http://www.w3.org/2000/svg",
    width: "1em",
    height: "1em",
    viewBox: "0 0 49 49",
    fill: "none"
  }, props), _path$n || (_path$n = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M17.871 44.612a3.38 3.38 0 002.39-5.77 3.38 3.38 0 00-2.39-.99 3.38 3.38 0 00-2.39.99 3.38 3.38 0 00-.99 2.39 3.384 3.384 0 003.38 3.38zm0-5.21a1.83 1.83 0 11-1.293 3.124 1.83 1.83 0 01-.536-1.294 1.836 1.836 0 011.83-1.834v.005zm9.142 5.21a3.38 3.38 0 003.38-3.38 3.38 3.38 0 00-.99-2.39 3.38 3.38 0 00-2.39-.99 3.38 3.38 0 100 6.76zm0-5.21a1.829 1.829 0 11-1.294.535 1.835 1.835 0 011.294-.54v.006zm-21.67 1.825a3.38 3.38 0 103.38-3.38 3.386 3.386 0 00-2.388.992 3.386 3.386 0 00-.992 2.388zm3.38-1.83a1.831 1.831 0 011.294 3.124 1.83 1.83 0 01-1.294.536 1.83 1.83 0 01-1.294-.536 1.828 1.828 0 01-.535-1.294 1.834 1.834 0 01.537-1.292 1.837 1.837 0 011.292-.537z",
    fill: "currentColor"
  })), _path2$c || (_path2$c = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M44.833 18.502h-.119L39.34 5.97a4.136 4.136 0 10-7.448-.078l-9.173 8.202h-8.75a.775.775 0 00-.775.775v4.651H6.568a.775.775 0 00-.775.775V30.72c0 .428.348.775.775.775h3.293v2.507H7.235a7.235 7.235 0 100 14.47h21.273a7.236 7.236 0 000-14.47h-2.626v-2.506h2.75a.775.775 0 00.775-.776v-5.757a.77.77 0 00-.259-.61l-4.904-9.54 8.563-7.634a4.137 4.137 0 002.682 1.119c1 .036 1.979-.292 2.755-.922l4.936 11.499a4.134 4.134 0 00.119 7.649l-6.641 7.07a.775.775 0 00.49 1.302l8.197.811h.28a2.843 2.843 0 002.842-2.842V24.75a.718.718 0 000-.088 4.134 4.134 0 00-3.618-6.166l-.016.006zm-21.887-2.857l4.372 8.542H17.262L14.745 20.1v-4.455h8.201zm11.251 25.582a5.686 5.686 0 01-5.685 5.685H7.235a5.685 5.685 0 110-11.37h21.273a5.686 5.686 0 015.674 5.69l.015-.005zm-9.85-7.235h-12.92v-2.497h12.92v2.497zm3.55-4.057H7.345v-8.848h6.202l2.615 4.279c.14.23.391.372.662.372h11.06l.015 4.197zM35.65 6.72a2.584 2.584 0 01-1.827-.757 2.584 2.584 0 010-3.655 2.583 2.583 0 011.827-.757 2.584 2.584 0 012.584 2.584c.003.69-.27 1.352-.758 1.84a2.586 2.586 0 01-1.841.755l.015-.01zm6.59 15.932c0-.685.272-1.342.756-1.827a2.582 2.582 0 011.827-.757 2.583 2.583 0 012.584 2.584 2.585 2.585 0 01-.756 1.827 2.584 2.584 0 01-1.828.757 2.584 2.584 0 01-2.584-2.584zm4.253 11.184a1.276 1.276 0 01-.992.33l-6.636-.661 6.305-6.719a4.136 4.136 0 001.762-.548v6.636c0 .37-.161.72-.44.962z",
    fill: "currentColor"
  })));
}

var _path$m, _path2$b;

function _extends$n() { _extends$n = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$n.apply(this, arguments); }

function SvgFilterApplied(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$n({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$m || (_path$m = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M10.593 7.346l1.754-1.753a2.01 2.01 0 012.842 0l.695.695L19.172 3H4c-.553 0-1 .442-1 .99v2.556c0 .525.211 1.029.586 1.4L9.707 14A.984.984 0 0110 14.7v6.31c0 .735.782 1.213 1.447.884l2-.989A.988.988 0 0014 20.02V14.7c0-.262.105-.514.293-.7l.056-.055-3.756-3.756a2.01 2.01 0 010-2.843z"
  })), _path2$b || (_path2$b = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M22.76 5.775a.01.01 0 000-.014l-1.753-1.754a.01.01 0 00-.014 0L15.89 9.11a.01.01 0 01-.014 0l-2.102-2.102a.01.01 0 00-.014 0l-1.754 1.754a.01.01 0 000 .014l3.87 3.87a.01.01 0 00.014 0l6.87-6.87z"
  })));
}

var _path$l;

function _extends$m() { _extends$m = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$m.apply(this, arguments); }

function SvgForward(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$m({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$l || (_path$l = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M11.834 19.413c-.186 0-.328-.062-.427-.186a.705.705 0 01-.148-.454v-3.74h-.39c-1.373 0-2.563.134-3.572.4-1.002.26-1.846.674-2.532 1.243-.687.569-1.225 1.317-1.615 2.245-.08.18-.167.306-.26.38a.532.532 0 01-.343.112.52.52 0 01-.39-.176C2.054 19.125 2 18.97 2 18.773c0-1.757.21-3.266.63-4.528.428-1.268 1.03-2.303 1.81-3.107a7.19 7.19 0 012.802-1.791c1.088-.39 2.297-.585 3.627-.585h.39V4.94c0-.173.05-.321.148-.445.1-.124.241-.186.427-.186.124 0 .238.031.343.093.112.056.214.13.306.223l6.847 6.345c.223.21.362.384.418.52.055.13.083.254.083.371a.926.926 0 01-.093.38c-.055.13-.191.297-.408.501l-6.847 6.365c-.092.08-.194.151-.306.213a.665.665 0 01-.343.093z"
  })));
}

var _path$k;

function _extends$l() { _extends$l = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$l.apply(this, arguments); }

function SvgHistory(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$l({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$k || (_path$k = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M12 3.75c-2.228 0-4.25.884-5.734 2.319l-.97-.969a.707.707 0 00-1.204.436L3.77 9.092a.707.707 0 00.768.768l3.556-.323a.707.707 0 00.436-1.205l-.94-.94A6.379 6.379 0 115.621 12a.936.936 0 10-1.871 0A8.25 8.25 0 1012 3.75zm0 3.743a.936.936 0 00-.936.936V13c0 .323.166.623.44.793l2.743 1.715a.936.936 0 00.992-1.587l-2.303-1.44V8.43A.936.936 0 0012 7.493z"
  })));
}

var _path$j, _path2$a, _path3$2;

function _extends$k() { _extends$k = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$k.apply(this, arguments); }

function SvgLayersOutline(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$k({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$j || (_path$j = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M20.605 8.21a.25.25 0 000-.437l-8.484-4.706a.25.25 0 00-.242 0L3.394 7.773a.25.25 0 000 .438l8.485 4.705a.25.25 0 00.242 0l7.239-4.018 1.245-.688zm-8.726-3.03a.25.25 0 01.242 0l4.673 2.592a.25.25 0 010 .437l-4.673 2.594a.25.25 0 01-.243 0L7.202 8.211a.25.25 0 010-.437L11.88 5.18z"
  })), _path2$a || (_path2$a = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M3.286 15.766a.25.25 0 00.023.41l8.57 5.244a.25.25 0 00.261 0l8.552-5.243a.25.25 0 00.023-.41L19.37 14.72l-7.245 4.181a.25.25 0 01-.25 0L4.62 14.73l-1.334 1.036z"
  })), _path3$2 || (_path3$2 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M3.286 11.534a.25.25 0 00.023.41l8.57 5.244a.25.25 0 00.261 0l8.552-5.243a.25.25 0 00.023-.41l-1.345-1.047-7.245 4.182a.25.25 0 01-.25 0L4.62 10.498l-1.334 1.036z"
  })));
}

var _path$i;

function _extends$j() { _extends$j = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$j.apply(this, arguments); }

function SvgLightning(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$j({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$i || (_path$i = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M6.12 12.347l7.958-10.3c.077-.1.238-.028.211.093l-1.64 7.43a.116.116 0 00.104.14l4.712.456c.47.046.7.58.402.937L8.815 21.957c-.08.096-.24.018-.207-.103l2.196-7.993a.116.116 0 00-.101-.145l-4.174-.45c-.457-.05-.685-.562-.41-.919z"
  })));
}

var _path$h, _path2$9;

function _extends$i() { _extends$i = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$i.apply(this, arguments); }

function SvgLock(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$i({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$h || (_path$h = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M13.7 15.2c0 .565-.276 1.066-.7 1.375v.925a1 1 0 11-2 0v-.925a1.7 1.7 0 112.7-1.375z"
  })), _path2$9 || (_path2$9 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M6 8a6 6 0 1112 0v2h1a1 1 0 011 1v10a1 1 0 01-1 1H5a1 1 0 01-1-1V11a1 1 0 011-1h1V8zm10 0v2H8V8a4 4 0 118 0zM6 12v8h12v-8H6z"
  })));
}

var _path$g, _path2$8;

function _extends$h() { _extends$h = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$h.apply(this, arguments); }

function SvgMapLocationPin(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$h({
    xmlns: "http://www.w3.org/2000/svg",
    width: "1em",
    height: "1em",
    viewBox: "0 0 31 45",
    fill: "none"
  }, props), _path$g || (_path$g = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M18.692 37.896c4.084-5.575 11.465-16.45 11.465-22.26C30.157 7.556 23.581.98 15.5.98 7.42.98.843 7.555.843 15.636c0 5.81 7.38 16.684 11.465 22.26-3.758.185-9.854.825-9.854 3.02 0 3.018 11.71 3.104 13.046 3.104 1.337 0 13.048-.086 13.048-3.102 0-2.196-6.097-2.836-9.855-3.022h-.001zM15.5 2.425c7.286 0 13.213 5.927 13.213 13.212 0 6.492-10.618 20.722-13.213 24.103-2.596-3.38-13.214-17.61-13.214-24.103 0-7.285 5.927-13.212 13.214-13.212zm0 40.15c-6.992 0-10.855-1.045-11.54-1.658.595-.525 3.682-1.419 9.391-1.614.88 1.171 1.472 1.923 1.583 2.063a.722.722 0 001.131 0c.112-.14.703-.891 1.583-2.063 5.71.195 8.797 1.089 9.394 1.614-.687.614-4.551 1.659-11.542 1.659z",
    fill: "currentColor"
  })), _path2$8 || (_path2$8 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M23.11 15.636c0-4.195-3.414-7.608-7.61-7.608s-7.61 3.413-7.61 7.608c0 4.196 3.414 7.61 7.61 7.61s7.61-3.414 7.61-7.61zm-13.775 0A6.172 6.172 0 0115.5 9.473c3.4 0 6.165 2.765 6.165 6.165 0 3.399-2.765 6.164-6.165 6.164a6.172 6.172 0 01-6.165-6.165z",
    fill: "currentColor"
  })));
}

var _path$f, _path2$7, _path3$1;

function _extends$g() { _extends$g = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$g.apply(this, arguments); }

function SvgPolygonalSelector(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$g({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$f || (_path$f = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M17.583 4.646c.144.51.658.816 1.17.71l-.118.667a1.005 1.005 0 001.98.348l.368-2.077a1.099 1.099 0 00-.043-.55.998.998 0 00-.313-.449 1.002 1.002 0 00-.479-.266 1.11 1.11 0 00-.554.012l-1.315.37c-.535.149-.847.702-.696 1.235zM5.666 9.028a1.002 1.002 0 10-.545-1.93l-1.316.369c-.18.05-.34.144-.469.268a.997.997 0 00-.272.46.994.994 0 00-.047.531 1.1 1.1 0 00.224.49l.736.918c.347.432.98.503 1.414.157a1 1 0 00.257-1.258l.018-.005zM19.88 10.524a1.005 1.005 0 00-1.98-.348l-.282 1.589-.593.243a1 1 0 00-.547 1.308c.211.512.8.756 1.313.545l1.053-.432c.087-.036.168-.082.241-.136.076-.057.142-.123.197-.194.059-.07.109-.147.148-.234a1.1 1.1 0 00.082-.264l.368-2.077z"
  })), _path2$7 || (_path2$7 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M9.967 14.387a1.008 1.008 0 00-1.414-.157 1 1 0 00-.331 1.083 2.503 2.503 0 001.24 4.685c.847 0 1.596-.417 2.051-1.056A2.512 2.512 0 0013.987 21a2.51 2.51 0 002.514-2.506H14.49a.502.502 0 01-1.006 0c0-.61-.186-1.247-.643-1.741a2.306 2.306 0 00-.953-.617.999.999 0 00.13-.99 1.007 1.007 0 00-1.313-.545l-.426.175-.312-.39zm-.505 2.604a.502.502 0 10-.002 1.004.502.502 0 00.002-1.004z"
  })), _path3$1 || (_path3$1 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M13.56 6.815l2.632-.738-.545-1.93-2.632.738.545 1.93zM8.297 8.29l2.631-.738-.544-1.93-2.631.739.544 1.929zM8.494 12.552L7.02 10.717 5.45 11.97l1.472 1.835 1.57-1.252zM14.918 12.873l-2.106.864.765 1.853 2.107-.864-.766-1.854z"
  })));
}

var _path$e, _path2$6;

function _extends$f() { _extends$f = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$f.apply(this, arguments); }

function SvgProjectsList(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$f({
    width: "1em",
    height: "1em",
    viewBox: "0 0 29 28",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$e || (_path$e = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M23.667 24.5H5a2.333 2.333 0 01-2.333-2.333V5.833A2.333 2.333 0 015 3.5h18.667A2.333 2.333 0 0126 5.833v16.334a2.333 2.333 0 01-2.333 2.333zM5 9.333v12.834h18.667V9.333H5zm0-3.5V7h18.667V5.833H5zm16.917 14.584h-7.584v-9.334h7.584v9.334z"
  })), _path2$6 || (_path2$6 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M6.75 13.417h5.833v-2.334H6.75v2.334zM6.75 16.917h5.833v-2.334H6.75v2.334zM6.75 20.417h5.833v-2.334H6.75v2.334z"
  })));
}

var _path$d, _path2$5;

function _extends$e() { _extends$e = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$e.apply(this, arguments); }

function SvgRoute(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$e({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$d || (_path$d = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M8.965 20.944c.404-.353.798-.718 1.181-1.093a.752.752 0 01-.52 1.093c-.082.016-.166.031-.253.046a.748.748 0 01-.408-.046zM20.779 6.804a.75.75 0 10-.558 1.392c.085.034.166.067.243.1a.75.75 0 00.582-1.383 16.456 16.456 0 00-.267-.11zM22.274 10.344a.75.75 0 10-1.051-1.07l-.19.183a.53.53 0 01-.073.056.75.75 0 00.824 1.253c.097-.063.193-.137.284-.223.07-.066.14-.133.206-.199zM19.052 11.77a.75.75 0 10-.498-1.415 8.246 8.246 0 00-.468.18.75.75 0 10.582 1.382c.12-.05.248-.099.384-.147zM18.032 12.935a.75.75 0 10-1.305.739c.045.08.095.161.149.242.062.093.128.178.199.256a.75.75 0 101.115-1.003.751.751 0 01-.066-.085 2.47 2.47 0 01-.092-.149zM20.236 15.207a.75.75 0 00-1.492.152c.003.04.006.086.006.141 0 .061 0 .12-.002.176a.75.75 0 101.5.04l.002-.216c0-.1-.005-.198-.014-.293zM18.636 18.27a.75.75 0 10-.476-1.423 2.966 2.966 0 01-.3.081.75.75 0 00.322 1.465c.156-.034.308-.074.454-.123zM15.789 18.72a.75.75 0 10-.187-1.489 57.908 57.908 0 00-.452.061.75.75 0 10.225 1.483 9.985 9.985 0 01.414-.056zM12.927 19.608a.75.75 0 00-.725-1.314l-.342.193-.081.046a.75.75 0 10.739 1.305l.085-.048c.11-.063.216-.123.324-.182z"
  })), _path2$5 || (_path2$5 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M8.9 21a27.896 27.896 0 01-3.25-3.334C4.164 15.86 2.4 13.174 2.4 10.613c-.001-2.674 1.583-5.086 4.012-6.11a6.421 6.421 0 017.086 1.435 6.624 6.624 0 011.902 4.675c0 2.561-1.764 5.248-3.25 7.053A27.897 27.897 0 018.9 21zm0-15.108c-2.563.003-4.64 2.115-4.643 4.721 0 1.101.49 3.008 2.818 5.841A26.202 26.202 0 008.9 18.451a26.367 26.367 0 001.826-1.994c2.327-2.837 2.817-4.744 2.817-5.844-.003-2.606-2.08-4.718-4.643-4.721zm0 7.554c-1.539 0-2.786-1.268-2.786-2.833 0-1.564 1.247-2.832 2.786-2.832 1.538 0 2.786 1.268 2.786 2.832 0 .752-.294 1.472-.816 2.003a2.762 2.762 0 01-1.97.83z"
  })));
}

var _path$c;

function _extends$d() { _extends$d = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$d.apply(this, arguments); }

function SvgRuler(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$d({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$c || (_path$c = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M6.344 12l2.121 2.122m.707-4.95l2.121 2.12M12 6.344l2.122 2.122M5.07 13.273l8.202-8.203c.792-.792 1.188-1.188 1.645-1.336a2 2 0 011.236 0c.456.148.852.544 1.643 1.335L18.93 6.2c.792.792 1.188 1.19 1.337 1.646.13.402.13.834 0 1.235-.149.457-.545.853-1.337 1.645l-8.202 8.203c-.792.792-1.189 1.188-1.645 1.336-.402.13-.834.13-1.235 0-.457-.148-.854-.544-1.646-1.336l-1.133-1.133c-.79-.791-1.187-1.187-1.335-1.643a2 2 0 010-1.236c.148-.457.545-.853 1.337-1.645z",
    stroke: "currentColor",
    strokeWidth: 2,
    strokeLinecap: "round",
    strokeLinejoin: "round"
  })));
}

var _path$b, _path2$4;

function _extends$c() { _extends$c = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$c.apply(this, arguments); }

function SvgSortDown(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$c({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$b || (_path$b = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M6.3 15.021a.01.01 0 01-.018.007l-1.828-1.897a.01.01 0 00-.014 0l-1.437 1.362a.01.01 0 000 .014l4.289 4.494a.01.01 0 00.014 0l4.29-4.494a.01.01 0 000-.014L10.2 13.131a.01.01 0 00-.014 0l-1.87 1.898a.01.01 0 01-.017-.007V5.01A.01.01 0 008.29 5H6.31a.01.01 0 00-.01.01v10.011zM10.3 5.01V7h10.99a.01.01 0 00.01-.01V5.01a.01.01 0 00-.01-.01H10.3v.01z"
  })), _path2$4 || (_path2$4 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M10.3 9v2h8.99a.01.01 0 00.01-.01V9.01a.01.01 0 00-.01-.01H10.3zM17.29 13h-4.361l.063.062A2.01 2.01 0 0113.537 15h3.753a.01.01 0 00.01-.01v-1.98a.01.01 0 00-.01-.01zM11.981 17l-1.909 2h5.314a.01.01 0 00.01-.01v-1.98a.01.01 0 00-.01-.01h-3.405z"
  })));
}

var _path$a, _path2$3;

function _extends$b() { _extends$b = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$b.apply(this, arguments); }

function SvgSortUp(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$b({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$a || (_path$a = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M10.3 13v2h6.99a.01.01 0 00.01-.01v-1.98a.01.01 0 00-.01-.01H10.3zM10.064 5l1.91 2h9.315a.01.01 0 00.01-.01V5.01a.01.01 0 00-.01-.01H10.064zM19.29 9h-5.756a2.01 2.01 0 01-.563 1.967l-.035.033h6.353a.01.01 0 00.01-.01V9.01a.01.01 0 00-.01-.01zM10.3 17v2h5.086a.01.01 0 00.01-.01v-1.98a.01.01 0 00-.01-.01H10.3z"
  })), _path2$3 || (_path2$3 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M8.3 8.987a.01.01 0 01.016-.007l1.828 1.897a.01.01 0 00.014 0l1.437-1.362a.01.01 0 000-.014L7.307 5.008a.01.01 0 00-.014 0L3.002 9.5a.01.01 0 00.001.014l1.395 1.362a.01.01 0 00.014 0l1.87-1.897a.01.01 0 01.017.007v10.011c0 .006.005.01.01.01h1.98a.01.01 0 00.01-.01V8.987z"
  })));
}

var _circle, _path$9;

function _extends$a() { _extends$a = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$a.apply(this, arguments); }

function SvgSpinner(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$a({
    width: "1em",
    height: "1em",
    viewBox: "0 0 36 36",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _circle || (_circle = /*#__PURE__*/React__namespace.createElement("circle", {
    cx: 18,
    cy: 18,
    r: 14,
    stroke: "#ECECEC",
    strokeOpacity: 0.6,
    strokeWidth: 4
  })), _path$9 || (_path$9 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M31.846 20.346c1.065.18 1.794 1.196 1.485 2.231a16.002 16.002 0 01-28.376 4.688A16 16 0 0118.73 2.017c1.08.049 1.798 1.072 1.617 2.137-.18 1.065-1.193 1.766-2.274 1.76a12.087 12.087 0 1011.364 16c.35-1.023 1.345-1.748 2.41-1.568z",
    fill: "currentColor"
  })));
}

var _path$8, _path2$2;

function _extends$9() { _extends$9 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$9.apply(this, arguments); }

function SvgTarget(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$9({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$8 || (_path$8 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M12 14.5a2.5 2.5 0 100-5 2.5 2.5 0 000 5z"
  })), _path2$2 || (_path2$2 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M11 3h2v2.576A6.505 6.505 0 0118.424 11H21v2h-2.576A6.505 6.505 0 0113 18.424V21h-2v-2.576A6.505 6.505 0 015.576 13H3v-2h2.576A6.505 6.505 0 0111 5.576V3zm1 13.518a4.518 4.518 0 110-9.036 4.518 4.518 0 010 9.036z"
  })));
}

var _path$7;

function _extends$8() { _extends$8 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$8.apply(this, arguments); }

function SvgTicket(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$8({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$7 || (_path$7 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M2.586 13.192L13.192 2.586l3.772 3.771-.707.707c-.228.227-.248.636.029.913.277.278.686.257.913.03l.707-.707 3.772 3.771L11.07 21.678 7.3 17.906l.707-.707c.227-.227.248-.636-.03-.913-.277-.277-.686-.257-.913-.03l-.707.708-3.771-3.772zm2.828 0l1.084 1.084a2.674 2.674 0 012.894.596c.777.777.991 1.928.595 2.893l1.084 1.084 7.778-7.778-1.084-1.084a2.673 2.673 0 01-2.893-.595 2.674 2.674 0 01-.596-2.894l-1.084-1.084-7.778 7.778z"
  })));
}

var _path$6;

function _extends$7() { _extends$7 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$7.apply(this, arguments); }

function SvgTilde(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$7({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$6 || (_path$6 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M9.74 11.921c-.497-.053-.91.09-1.21.441-.364.428-.53.811-.53 1.138H6c0-.998.492-1.83 1.008-2.435.796-.934 1.902-1.245 2.947-1.132 1.021.11 2.01.619 2.752 1.36.453.453 1.03.73 1.552.786.498.053.91-.09 1.21-.441.365-.428.531-.812.531-1.138h2c0 .998-.492 1.83-1.008 2.435-.796.934-1.902 1.245-2.947 1.132-1.021-.11-2.01-.619-2.752-1.36-.453-.453-1.03-.73-1.552-.786z"
  })));
}

var _path$5;

function _extends$6() { _extends$6 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$6.apply(this, arguments); }

function SvgUploading(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$6({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$5 || (_path$5 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M6.75 17.25c-.615.615-1.626.622-2.125-.091a9 9 0 01-.446-.706c-.43-.756.008-1.666.828-1.958l.263-.095c.674-.24 1.404.123 1.786.727.382.605.397 1.42-.108 1.925l-.198.198zM10.144 4.81c-.217-.842.29-1.715 1.158-1.783A9 9 0 118.848 20.43c-.814-.305-1.061-1.285-.62-2.034.442-.75 1.408-.975 2.248-.748a5.85 5.85 0 101.621-11.497c-.87-.015-1.735-.498-1.953-1.34z"
  })));
}

var _path$4, _path2$1, _path3;

function _extends$5() { _extends$5 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$5.apply(this, arguments); }

function SvgUrbint(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$5({
    width: "1em",
    height: "1em",
    viewBox: "0 0 30 30",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$4 || (_path$4 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M15 12.877a2.123 2.123 0 100 4.246 2.123 2.123 0 000-4.246zM10.896 15a4.104 4.104 0 118.208 0 4.104 4.104 0 01-8.208 0z",
    fill: "currentColor"
  })), _path2$1 || (_path2$1 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M15 7.5a7.5 7.5 0 100 15 7.5 7.5 0 000-15zM5.519 15a9.481 9.481 0 1118.962 0A9.481 9.481 0 015.52 15z",
    fill: "currentColor"
  })), _path3 || (_path3 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M15 1.981C7.81 1.981 1.981 7.81 1.981 15S7.81 28.019 15 28.019 28.019 22.19 28.019 15 22.19 1.981 15 1.981zM0 15C0 6.716 6.716 0 15 0c8.284 0 15 6.716 15 15 0 8.284-6.716 15-15 15-8.284 0-15-6.716-15-15z",
    fill: "currentColor"
  })));
}

var _path$3;

function _extends$4() { _extends$4 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$4.apply(this, arguments); }

function SvgVideo(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$4({
    width: "1em",
    height: "1em",
    viewBox: "0 0 24 24",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path$3 || (_path$3 = /*#__PURE__*/React__namespace.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M5 4.5a3 3 0 00-3 3v9a3 3 0 003 3h14a3 3 0 003-3v-9a3 3 0 00-3-3H5zM5 6a1.5 1.5 0 00-1.5 1.5v1A.5.5 0 004 9h1.5a.5.5 0 00.5-.5v-2a.5.5 0 00-.5-.5H5zm4.862 8.82a.25.25 0 01-.362-.225v-5.19a.25.25 0 01.362-.224l5.19 2.595a.25.25 0 010 .448l-5.19 2.595zM18 6.5a.5.5 0 01.5-.5h.5a1.5 1.5 0 011.5 1.5v1a.5.5 0 01-.5.5h-1.5a.5.5 0 01-.5-.5v-2zm-14 4a.5.5 0 00-.5.5v2a.5.5 0 00.5.5h1.5A.5.5 0 006 13v-2a.5.5 0 00-.5-.5H4zm14 .5a.5.5 0 01.5-.5H20a.5.5 0 01.5.5v2a.5.5 0 01-.5.5h-1.5a.5.5 0 01-.5-.5v-2zM4 15a.5.5 0 00-.5.5v1A1.5 1.5 0 005 18h.5a.5.5 0 00.5-.5v-2a.5.5 0 00-.5-.5H4zm14 .5a.5.5 0 01.5-.5H20a.5.5 0 01.5.5v1A1.5 1.5 0 0119 18h-.5a.5.5 0 01-.5-.5v-2z"
  })));
}

var _g, _defs;

function _extends$3() { _extends$3 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$3.apply(this, arguments); }

function SvgWarningCircle(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$3({
    width: "1em",
    height: "1em",
    viewBox: "0 0 22 22",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _g || (_g = /*#__PURE__*/React__namespace.createElement("g", {
    filter: "url(#warning_circle_svg__filter0_i_4329_60808)"
  }, /*#__PURE__*/React__namespace.createElement("rect", {
    width: 22,
    height: 22,
    rx: 11
  }), /*#__PURE__*/React__namespace.createElement("path", {
    d: "M9 19h3.2v-3.2H9V19zM9 12.6h3.2V3H9v9.6z",
    fill: "#fff"
  }))), _defs || (_defs = /*#__PURE__*/React__namespace.createElement("defs", null, /*#__PURE__*/React__namespace.createElement("filter", {
    id: "warning_circle_svg__filter0_i_4329_60808",
    x: 0,
    y: 0,
    width: 22,
    height: 22,
    filterUnits: "userSpaceOnUse",
    colorInterpolationFilters: "sRGB"
  }, /*#__PURE__*/React__namespace.createElement("feFlood", {
    floodOpacity: 0,
    result: "BackgroundImageFix"
  }), /*#__PURE__*/React__namespace.createElement("feBlend", {
    "in": "SourceGraphic",
    in2: "BackgroundImageFix",
    result: "shape"
  }), /*#__PURE__*/React__namespace.createElement("feColorMatrix", {
    "in": "SourceAlpha",
    values: "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0",
    result: "hardAlpha"
  }), /*#__PURE__*/React__namespace.createElement("feOffset", null), /*#__PURE__*/React__namespace.createElement("feGaussianBlur", {
    stdDeviation: 1
  }), /*#__PURE__*/React__namespace.createElement("feComposite", {
    in2: "hardAlpha",
    operator: "arithmetic",
    k2: -1,
    k3: 1
  }), /*#__PURE__*/React__namespace.createElement("feColorMatrix", {
    values: "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.07 0"
  }), /*#__PURE__*/React__namespace.createElement("feBlend", {
    in2: "shape",
    result: "effect1_innerShadow_4329_60808"
  })))));
}

var _path$2;

function _extends$2() { _extends$2 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$2.apply(this, arguments); }

function SvgWorkerHelmet(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$2({
    xmlns: "http://www.w3.org/2000/svg",
    width: "1em",
    height: "1em",
    viewBox: "0 0 43 29",
    fill: "none"
  }, props), _path$2 || (_path$2 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M39.417 21.665v-.716A17.902 17.902 0 0035.045 9.22a17.639 17.639 0 00-2.209-2.142 17.892 17.892 0 00-6.567-3.392 4.986 4.986 0 00-9.531 0 17.891 17.891 0 00-6.565 3.39A17.533 17.533 0 009.02 8.102a19.214 19.214 0 00-1.065 1.12 17.902 17.902 0 00-4.372 11.727v.716a3.583 3.583 0 000 7.167h35.834a3.583 3.583 0 000-7.167zm-1.433 0H26.517V5.25a16.469 16.469 0 014.923 2.58l-1.77 3.06a2.168 2.168 0 00-.21 1.632c.148.551.506 1.021.998 1.31a2.15 2.15 0 001.634.212 2.148 2.148 0 001.305-1.007l1.199-2.075a16.458 16.458 0 013.388 9.987v.717zm-5.22-12.732c.188.175.367.359.546.539.102.105.203.215.302.321l-1.46 2.528a.716.716 0 01-.435.369.716.716 0 01-.803-1.086l1.649-2.851c.065.06.135.118.202.18zM21.5 1.598c.818.003 1.61.285 2.247.799A3.596 3.596 0 0125 4.425c.055.248.083.502.084.757v16.483h-7.167V5.182a3.52 3.52 0 01.084-.76 3.596 3.596 0 011.254-2.026 3.596 3.596 0 012.245-.798zM9.692 9.471c.178-.182.358-.363.542-.537.068-.063.138-.121.206-.183l1.645 2.85a.718.718 0 01-.234 1.023.716.716 0 01-.568.065.716.716 0 01-.435-.372l-1.46-2.524c.1-.109.2-.216.304-.322zM5.017 20.948a16.46 16.46 0 013.39-9.987l1.197 2.073a2.15 2.15 0 001.304 1.008 2.149 2.149 0 001.634-.212h.005c.49-.289.847-.759.994-1.309a2.166 2.166 0 00-.21-1.63l-1.768-3.06a16.473 16.473 0 014.92-2.582v16.416H5.018v-.717zm34.4 6.45H3.584a2.15 2.15 0 010-4.3h35.833a2.15 2.15 0 010 4.3z",
    fill: "currentColor"
  })));
}

var _path$1;

function _extends$1() { _extends$1 = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends$1.apply(this, arguments); }

function SvgWrench(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends$1({
    xmlns: "http://www.w3.org/2000/svg",
    width: "1em",
    height: "1em",
    viewBox: "0 0 25 25",
    fill: "none"
  }, props), _path$1 || (_path$1 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M5.132 25h-.328c1.17-.133 2.076-.754 2.878-1.569.894-.907 1.23-1.989.908-3.24-.106-.411-.018-.687.27-.995 2.757-2.929 5.576-5.793 8.553-8.492.73-.663 1.481-1.301 2.218-1.956.122-.108.242-.141.39-.081 1.794.73 3.507-.312 4.309-1.688.438-.752.645-1.572.67-2.444.006-.216-.108-.366-.225-.514-.085-.107-.198-.068-.304 0-.652.417-1.309.83-1.96 1.248-.558.358-1.063.323-1.533-.148-.403-.405-.703-.88-.804-1.46a1.257 1.257 0 01.579-1.314c.616-.397 1.228-.8 1.842-1.202.187-.123.193-.2.09-.403-.095-.191-.24-.297-.43-.375-1.589-.65-3.05-.433-4.408.619-.501.388-.968.797-1.254 1.381-.362.742-.386 1.521-.23 2.312.18.906.193.684-.371 1.289-3.021 3.24-6.145 6.373-9.461 9.303-.394.348-.797.686-1.19 1.034-.107.094-.214.109-.345.07-2.096-.629-3.982.368-4.702 2.488-.108.318-.197.642-.294.963v.47c.05.184.105.367.151.551.064.255.157.3.375.161.665-.42 1.33-.839 1.99-1.264.483-.31.96-.299 1.395.077.44.38.74.86.888 1.43a1.33 1.33 0 01-.585 1.463c-.625.368-1.22.789-1.83 1.185-.106.07-.18.16-.126.283.083.184.167.377.389.441L3.942 25",
    fill: "currentColor"
  })));
}

var _path, _path2;

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function SvgMultilocation(props) {
  return /*#__PURE__*/React__namespace.createElement("svg", _extends({
    width: "1em",
    height: "1em",
    viewBox: "0 0 16 16",
    fill: "currentColor",
    xmlns: "http://www.w3.org/2000/svg"
  }, props), _path || (_path = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M5.2 13.6A18.618 18.618 0 013 11.403C1.994 10.214.8 8.444.8 6.757a4.354 4.354 0 012.716-4.025 4.433 4.433 0 014.796.945A4.304 4.304 0 019.6 6.757c0 1.687-1.195 3.457-2.2 4.646A18.617 18.617 0 015.2 13.6zm0-8.71a1.89 1.89 0 00-1.633.934 1.85 1.85 0 000 1.866c.337.577.96.933 1.633.933a1.876 1.876 0 001.885-1.866c0-1.03-.844-1.866-1.885-1.866z"
  })), _path2 || (_path2 = /*#__PURE__*/React__namespace.createElement("path", {
    d: "M8 11.329l.06.074c.644.792 1.351 1.527 2.114 2.197.762-.67 1.469-1.405 2.113-2.197.966-1.189 2.113-2.959 2.113-4.646a4.394 4.394 0 00-1.237-3.08A4.136 4.136 0 008 3.02 4.331 4.331 0 019.68 4.96a1.76 1.76 0 01.494-.07c1 0 1.81.835 1.81 1.866 0 1.03-.81 1.866-1.81 1.866a1.76 1.76 0 01-.494-.07C9.276 9.59 8.608 10.57 8 11.327z"
  })));
}

var ArrowheadOutline = SvgArrowheadOutline;
var Asterisk = SvgAsterisk;
var Attachment = SvgAttachment;
var Calculator = SvgCalculator;
var Camera = SvgCamera;
var Circle = SvgCircle;
var CrossHammerWrench = SvgCrossHammerWrench;
var CrossLightning = SvgCrossLightning;
var Excavator = SvgExcavator;
var FilterApplied = SvgFilterApplied;
var Forward = SvgForward;
var History = SvgHistory;
var LayersOutline = SvgLayersOutline;
var Lightning = SvgLightning;
var Lock = SvgLock;
var MapLocationPin = SvgMapLocationPin;
var PolygonalSelector = SvgPolygonalSelector;
var ProjectsList = SvgProjectsList;
var Route = SvgRoute;
var Ruler = SvgRuler;
var SortDown = SvgSortDown;
var SortUp = SvgSortUp;
var Spinner = SvgSpinner;
var Target = SvgTarget;
var Ticket = SvgTicket;
var Tilde = SvgTilde;
var Uploading = SvgUploading;
var Urbint = SvgUrbint;
var Video = SvgVideo;
var WarningCircle = SvgWarningCircle;
var WorkerHelmet = SvgWorkerHelmet;
var Wrench = SvgWrench;
var MultiLocation = SvgMultilocation;

var Icon = function (_a) {
    var _b;
    var name = _a.name, className = _a.className, props = tslib.__rest(_a, ["name", "className"]);
    var ChildIcon;
    switch (name) {
        case 'arrowhead_outline':
            ChildIcon = ArrowheadOutline;
            break;
        case 'asterisk':
            ChildIcon = Asterisk;
            break;
        case 'attachment':
            ChildIcon = Attachment;
            break;
        case 'calculator':
            ChildIcon = Calculator;
            break;
        case 'camera':
            ChildIcon = Camera;
            break;
        case 'circle':
            ChildIcon = Circle;
            break;
        case 'cross_hammer_wrench':
            ChildIcon = CrossHammerWrench;
            break;
        case 'cross_lightning':
            ChildIcon = CrossLightning;
            break;
        case 'excavator':
            ChildIcon = Excavator;
            break;
        case 'filter_applied':
            ChildIcon = FilterApplied;
            break;
        case 'forward':
            ChildIcon = Forward;
            break;
        case 'history':
            ChildIcon = History;
            break;
        case 'layers_outline':
            ChildIcon = LayersOutline;
            break;
        case 'lightning':
            ChildIcon = Lightning;
            break;
        case 'lock':
            ChildIcon = Lock;
            break;
        case 'map_location_pin':
            ChildIcon = MapLocationPin;
            break;
        case 'polygonal_selector':
            ChildIcon = PolygonalSelector;
            break;
        case 'projects_list':
            ChildIcon = ProjectsList;
            break;
        case 'route':
            ChildIcon = Route;
            break;
        case 'sort_down':
            ChildIcon = SortDown;
            break;
        case 'sort_up':
            ChildIcon = SortUp;
            break;
        case 'spinner':
            ChildIcon = Spinner;
            break;
        case 'target':
            ChildIcon = Target;
            break;
        case 'ticket':
            ChildIcon = Ticket;
            break;
        case 'tilde':
            ChildIcon = Tilde;
            break;
        case 'uploading':
            ChildIcon = Uploading;
            break;
        case 'urbint':
            ChildIcon = Urbint;
            break;
        case 'video':
            ChildIcon = Video;
            break;
        case 'warning_circle':
            ChildIcon = WarningCircle;
            break;
        case 'worker_helmet':
            ChildIcon = WorkerHelmet;
            break;
        case 'wrench':
            ChildIcon = Wrench;
            break;
        case 'ruler':
            ChildIcon = Ruler;
            break;
        case 'multi_location':
            ChildIcon = MultiLocation;
            break;
    }
    return (jsxRuntime.jsx("i", tslib.__assign({ className: cx((_b = {}, _b["ci-".concat(name)] = !ChildIcon, _b), className), "aria-hidden": true }, props, { children: ChildIcon ? jsxRuntime.jsx(ChildIcon, {}) : undefined })));
};

var Badge$1 = function (_a) {
    var label = _a.label, title = _a.title, className = _a.className, iconStart = _a.iconStart, iconEnd = _a.iconEnd;
    var labelMargin = '';
    if (!iconStart && !iconEnd) {
        labelMargin = 'mx-1';
    }
    else if (!iconStart) {
        labelMargin += 'ml-1 ';
    }
    else if (!iconEnd) {
        labelMargin += 'mr-1';
    }
    return (jsxRuntime.jsx("div", tslib.__assign({ className: "flex items-center font-bold shadow-5 h-5 rounded-full p-0.5 w-max uppercase bg-brand-gray-20 ".concat(className), role: "note" }, { children: jsxRuntime.jsxs("span", tslib.__assign({ className: "flex items-center px-1", title: title }, { children: [iconStart && jsxRuntime.jsx(Icon, { name: iconStart }), label && jsxRuntime.jsx("span", tslib.__assign({ className: "text-tiny ".concat(labelMargin) }, { children: label })), iconEnd && jsxRuntime.jsx(Icon, { name: iconEnd })] })) })));
};

var Typography = function (_a) {
    var Component = _a.variant, htmlFor = _a.htmlFor, children = _a.children, props = tslib.__rest(_a, ["variant", "htmlFor", "children"]);
    return (jsxRuntime.jsx(Component, tslib.__assign({}, (htmlFor ? { htmlFor: htmlFor } : {}), props, { children: children })));
};

var PageHeading = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-primary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "h1", className: cx('text-page-heading font-page-heading', color, className) }, props, { children: children })));
};

var SectionHeading = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-primary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "h2", className: cx('text-section-heading font-section-heading', color, className) }, props, { children: children })));
};

var SectionSubheading = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-primary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "h4", className: cx('text-section-subheading font-section-subheading', color, className) }, props, { children: children })));
};

var Subheading = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-primary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "h3", className: cx('text-subheading font-subheading', color, className) }, props, { children: children })));
};

var ActionLabel = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-primary' : _b, children = _a.children, className = _a.className, props = tslib.__rest(_a, ["color", "children", "className"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "span", className: cx('text-action-label font-action-label', color, className) }, props, { children: children })));
};

var BodyText = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-primary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "p", className: cx('text-body-text font-body-text', color, className) }, props, { children: children })));
};

var ComponentLabel = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-secondary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "label", className: cx('text-component-label font-component-label', color, className) }, props, { children: children })));
};

var CaptionText = function (_a) {
    var _b = _a.color, color = _b === void 0 ? 'text-neutrals-secondary' : _b, className = _a.className, children = _a.children, props = tslib.__rest(_a, ["color", "className", "children"]);
    return (jsxRuntime.jsx(Typography, tslib.__assign({ variant: "p", className: cx('text-caption-text font-caption-text', color, className) }, props, { children: children })));
};

var BG_COLOR_CLASSES_MAP = {
    positive: {
        gray: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-neutral-secondary',
        },
        red: { backgroundColor: 'bg-badges-gray', textColor: 'text-badges-red' },
        orange: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-badges-orange',
        },
        blue: { backgroundColor: 'bg-badges-gray', textColor: 'text-badges-blue' },
        green: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-badges-green',
        },
        darkgray: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-badges-darkgray',
        },
        purple: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-badges-purple',
        },
        neon: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-badges-green-neon',
        },
    },
    negative: {
        gray: {
            backgroundColor: 'bg-badges-gray',
            textColor: 'text-neutral-secondary',
        },
        darkgray: {
            backgroundColor: 'bg-badges-darkgray',
            textColor: 'text-white',
        },
        red: { backgroundColor: 'bg-badges-red', textColor: 'text-white' },
        orange: { backgroundColor: 'bg-badges-orange', textColor: 'text-white' },
        blue: { backgroundColor: 'bg-badges-blue', textColor: 'text-white' },
        green: { backgroundColor: 'bg-badges-green', textColor: 'text-white' },
        purple: { backgroundColor: 'bg-badges-purple', textColor: 'text-white' },
        neon: { backgroundColor: 'bg-badges-green-neon', textColor: 'text-white' },
    },
    simple: {
        gray: { textColor: 'text-neutral-secondary' },
        darkgray: { textColor: 'text-badges-darkgray' },
        red: { textColor: 'text-badges-red' },
        orange: { textColor: 'text-badges-orange' },
        blue: { textColor: 'text-badges-blue' },
        green: { textColor: 'text-badges-green' },
        purple: { textColor: 'text-badges-purple' },
        neon: { textColor: 'text-badges-green-neon' },
    },
};
var Badge = function (_a) {
    var _b;
    var _c, _d;
    var _e = _a.testId, testId = _e === void 0 ? '' : _e, text = _a.text, className = _a.className, _f = _a.color, color = _f === void 0 ? 'gray' : _f, iconName = _a.iconName, _g = _a.type, type = _g === void 0 ? 'negative' : _g;
    return (jsxRuntime.jsxs("div", tslib.__assign({ "data-testid": testId, className: cx('py-0.5 min-w-[24px] justify-center rounded-3xl uppercase flex items-center gap-1', (_b = {},
            _b[BG_COLOR_CLASSES_MAP[type][color]
                .backgroundColor] = type !== 'simple',
            _b['px-1'] = type === 'simple' || !text,
            _b['px-2'] = type !== 'simple' && text,
            _b), className) }, { children: [iconName && (jsxRuntime.jsx(Icon, { "data-testid": "badge-icon", name: iconName, className: cx('text-base', (_c = BG_COLOR_CLASSES_MAP[type]) === null || _c === void 0 ? void 0 : _c[color].textColor) })), text && (jsxRuntime.jsx(ComponentLabel, tslib.__assign({ color: (_d = BG_COLOR_CLASSES_MAP[type]) === null || _d === void 0 ? void 0 : _d[color].textColor, className: "max-w-xs whitespace-nowrap overflow-hidden overflow-ellipsis" }, { children: text })))] })));
};

var createPortalAndAppendToBody = function (portalId) {
    var portalElement = document.createElement('div');
    if (portalId)
        portalElement.setAttribute('id', portalId);
    document.body.appendChild(portalElement);
    return portalElement;
};
var Portal = function (_a) {
    var _b = _a.id, id = _b === void 0 ? 'silica-portal' : _b, children = _a.children;
    var _c = React.useState(null), portalElement = _c[0], setPortalElement = _c[1];
    React.useLayoutEffect(function () {
        var element = document.getElementById(id);
        if (!element) {
            element = createPortalAndAppendToBody(id);
        }
        setPortalElement(element);
        return function () {
            if (element === null || element === void 0 ? void 0 : element.parentNode) {
                element.parentNode.removeChild(element);
            }
        };
    }, [id]);
    if (!portalElement) {
        return null;
    }
    return reactDom.createPortal(children, portalElement);
};

var Banner = function (_a) {
    var className = _a.className, _b = _a.dataTestId, dataTestId = _b === void 0 ? 'base-banner-component' : _b, children = _a.children;
    return (jsxRuntime.jsx("div", tslib.__assign({ className: cx('relative min-h-10 p-2 flex items-center', className), "data-testid": dataTestId }, { children: children })));
};

var styles$1 = {"Thumb":"Slider-module_Thumb__MCUYX"};

var Slider = function (_a) {
    var _b = _a.type, type = _b === void 0 ? 'single' : _b, _c = _a.min, min = _c === void 0 ? 0 : _c, _d = _a.max, max = _d === void 0 ? 100 : _d, value = _a.value, _e = _a.gap, gap = _e === void 0 ? 1 : _e, _f = _a.step, step = _f === void 0 ? 1 : _f, _g = _a.withInputs, withInputs = _g === void 0 ? true : _g, label = _a.label, _h = _a.readonly, readonly = _h === void 0 ? false : _h, _j = _a.disabled, disabled = _j === void 0 ? false : _j, onChange = _a.onChange;
    var _k = React.useState((value === null || value === void 0 ? void 0 : value[0]) || min), minVal = _k[0], setMinVal = _k[1];
    var _l = React.useState((value === null || value === void 0 ? void 0 : value[1]) || max), maxVal = _l[0], setMaxVal = _l[1];
    var minValRef = React.useRef(null);
    var maxValRef = React.useRef(null);
    var range = React.useRef(null);
    var sliderRef = React.useRef(null);
    var _m = React.useState(false), showMinThumbTooltip = _m[0], setShowMinThumbTooltip = _m[1];
    var _o = React.useState(false), showMaxThumbTooltip = _o[0], setShowMaxThumbTooltip = _o[1];
    var isRange = type === 'range';
    var tempMin = isRange ? minVal : min;
    var tempMax = isRange ? maxVal : max;
    var isMinValValid = minVal >= min && minVal <= tempMax;
    var isMaxValValid = maxVal >= tempMin && maxVal <= max;
    var getPercent = function (value) {
        return Math.round(((value - min) / (max - min)) * 100);
    };
    var calculatePosition = function (value) {
        var slider = sliderRef.current;
        if (!slider)
            return 0;
        var width = slider.offsetWidth - 24;
        var newPosition = ((value - min) / (max - min)) * width + 12;
        return newPosition;
    };
    var onMinThumbChanged = function (event) {
        var value = Math.min(+event.target.value, !isRange ? maxVal : maxVal - gap);
        setMinVal(Math.round(value / step) * step);
    };
    var onMaxThumbChanged = function (event) {
        var value = Math.max(+event.target.value, minVal + gap);
        setMaxVal(Math.round(value / step) * step);
    };
    React.useEffect(function () {
        setShowMinThumbTooltip(!withInputs);
        setShowMaxThumbTooltip(!withInputs && isRange);
    }, [withInputs, isRange]);
    React.useEffect(function () {
        if (!isRange) {
            var minPercent = getPercent(minVal);
            if (range.current) {
                range.current.style.left = '0';
                range.current.style.width = "".concat(minPercent <= max ? minPercent : max, "%");
            }
        }
        else {
            var tempMinValue = isMinValValid ? minVal : min;
            var tempMaxValue = isMaxValValid ? maxVal : max;
            var minPercent = getPercent(tempMinValue);
            var maxPercent = getPercent(tempMaxValue);
            if (range.current) {
                range.current.style.left = "".concat(minPercent, "%");
                range.current.style.width = "".concat(maxPercent - minPercent, "%");
            }
        }
        if (!isMinValValid || !isMaxValValid) {
            onChange({
                error: true,
                message: "Invalid value, please check the range values (min: ".concat(min, ", max: ").concat(max, ", currentMin: ").concat(minVal, ", currentMax: ").concat(maxVal, ")"),
            });
            return;
        }
        var values = isRange
            ? { min: minVal, max: maxVal }
            : { value: minVal };
        onChange(values);
    }, [minVal, maxVal, type, min, max]);
    return (jsxRuntime.jsxs("div", tslib.__assign({ className: cx('flex flex-col gap-3', {
            'gap-9': !withInputs,
        }) }, { children: [label && jsxRuntime.jsx(ComponentLabel, { children: label }), jsxRuntime.jsxs("div", tslib.__assign({ className: cx('flex justify-center items-center gap-2', {
                    'flex-row-reverse': !isRange,
                }) }, { children: [withInputs && (jsxRuntime.jsx("input", { type: "number", min: min, max: isRange ? maxVal : max, value: Number(minVal).toString(), step: step, readOnly: readonly, disabled: disabled, onChange: function (newValue) {
                            return setMinVal(Number(newValue.target.value.toString()));
                        }, className: cx('w-[62px] h-[34px] flex justify-center items-center text-center px-2 py-1 rounded bg-backgrounds-white border border-borders focus:border-actions-primary hover:shadow-elevation-10 outline-none', {
                            'text-neutrals-tertiary border-none': readonly,
                            'text-neutrals-disabled border-borders-disabled cursor-not-allowed': disabled,
                            'border-borders-danger focus:border-actions-danger-active': !isMinValValid,
                        }) })), jsxRuntime.jsxs("div", tslib.__assign({ className: "relative w-full", ref: sliderRef }, { children: [isMinValValid && (jsxRuntime.jsxs("div", tslib.__assign({ className: "group" }, { children: [jsxRuntime.jsx("input", { type: "range", "data-testid": "min-slider", min: min, max: max, value: minVal, ref: minValRef, onChange: onMinThumbChanged, className: cx("".concat(styles$1.Thumb, " absolute h-0 w-full outline-none pointer-events-none z-[25]"), {
                                            'z-30': minVal > max - 100,
                                        }), disabled: disabled }), showMinThumbTooltip && (jsxRuntime.jsx(BodyText, tslib.__assign({ className: cx('absolute z-20 -translate-x-1/2 -translate-y-9 hidden group-hover:block', {
                                            '!text-actions-primary-disabled': disabled,
                                        }), style: {
                                            left: "".concat(calculatePosition(minVal), "px"),
                                        }, color: "text-actions-primary" }, { children: minVal })))] }))), isRange && isMaxValValid && (jsxRuntime.jsxs("div", tslib.__assign({ className: "group" }, { children: [jsxRuntime.jsx("input", { type: "range", "data-testid": "max-slider", min: min, max: max, value: maxVal, step: step, ref: maxValRef, onChange: onMaxThumbChanged, className: cx(styles$1.Thumb, 'absolute h-0 w-full outline-none pointer-events-none z-20', {}), disabled: disabled }), showMaxThumbTooltip && (jsxRuntime.jsx(BodyText, tslib.__assign({ className: cx('absolute z-20 -translate-x-1/2 -translate-y-9 hidden group-hover:block', {
                                            '!text-actions-primary-disabled': disabled,
                                        }), style: {
                                            left: "".concat(calculatePosition(maxVal), "px"),
                                        }, color: "text-actions-primary" }, { children: maxVal })))] }))), jsxRuntime.jsx(jsxRuntime.Fragment, { children: jsxRuntime.jsxs("div", tslib.__assign({ className: "relative w-full" }, { children: [jsxRuntime.jsx("div", { className: "absolute w-full h-1 z-10 bg-backgrounds-overlay rounded-[3px]" }), jsxRuntime.jsx("div", { ref: range, className: cx('absolute h-1 z-10 bg-actions-primary', {
                                                'bg-actions-primary-disabled': disabled || readonly,
                                            }) }), jsxRuntime.jsxs("div", tslib.__assign({ className: "absolute w-full h-full top-full pt-[16px]" }, { children: [jsxRuntime.jsx(BodyText, tslib.__assign({ className: "absolute" }, { children: min })), jsxRuntime.jsx(BodyText, tslib.__assign({ className: "absolute right-0 text-right" }, { children: max }))] }))] })) })] })), withInputs && isRange && (jsxRuntime.jsx("input", { type: "number", min: minVal, max: max, value: Number(maxVal).toString(), readOnly: readonly, disabled: disabled, onChange: function (newValue) { return setMaxVal(+newValue.target.value); }, className: cx('w-[62px] h-[34px] flex justify-center items-center text-center px-2 py-1 rounded bg-backgrounds-white border border-borders focus:border-actions-primary hover:shadow-elevation-10 outline-none', {
                            'text-neutrals-tertiary border-none': readonly,
                            'text-neutrals-disabled border-borders-disabled cursor-not-allowed': disabled,
                            'border-borders-danger focus:border-actions-danger-active': !isMaxValValid,
                        }) }))] }))] })));
};

var styles = {"container":"BaseBanner-module_container__ToKUe","pointer-events-none":"BaseBanner-module_pointer-events-none__ohGOZ","absolute":"BaseBanner-module_absolute__pcuW3","relative":"BaseBanner-module_relative__JY3MR","top-full":"BaseBanner-module_top-full__-zqtT","right-0":"BaseBanner-module_right-0__ovm6P","left-1":"BaseBanner-module_left-1__th-lZ","top-1":"BaseBanner-module_top-1__a-kBm","right-4":"BaseBanner-module_right-4__1SY46","top-4":"BaseBanner-module_top-4__Nlr2U","z-10":"BaseBanner-module_z-10__7RUIp","z-20":"BaseBanner-module_z-20__43-zq","z-[25]":"BaseBanner-module_z---25--__SwU-K","z-30":"BaseBanner-module_z-30__Hfm-V","m-3":"BaseBanner-module_m-3__DEnzD","mx-3":"BaseBanner-module_mx-3__e10vk","mx-1":"BaseBanner-module_mx-1__a3uFT","mt-4":"BaseBanner-module_mt-4__dP6Ja","ml-1":"BaseBanner-module_ml-1__kGso7","mr-1":"BaseBanner-module_mr-1__VI4EG","flex":"BaseBanner-module_flex__TgFKm","table":"BaseBanner-module_table__Ybo-R","grid":"BaseBanner-module_grid__tBUDT","hidden":"BaseBanner-module_hidden__fAXq6","h-6":"BaseBanner-module_h-6__SVB4Q","h-10":"BaseBanner-module_h-10__YDOvg","h-[34px]":"BaseBanner-module_h---34px--__f-fZ3","h-0":"BaseBanner-module_h-0__kokD8","h-1":"BaseBanner-module_h-1__JrDtg","h-full":"BaseBanner-module_h-full__vIZFC","h-[26px]":"BaseBanner-module_h---26px--__luXHr","h-[18px]":"BaseBanner-module_h---18px--__P-P3n","h-48":"BaseBanner-module_h-48__tqZRZ","h-36":"BaseBanner-module_h-36__viXin","h-5":"BaseBanner-module_h-5__FxDSO","!w-max":"BaseBanner-module_--w-max__nWxuK","w-full":"BaseBanner-module_w-full__ALKdn","w-6":"BaseBanner-module_w-6__6WSZW","w-10":"BaseBanner-module_w-10__N0C6C","w-max":"BaseBanner-module_w-max__g28mG","w-[350px]":"BaseBanner-module_w---350px--__A2DQv","w-[62px]":"BaseBanner-module_w---62px--__s41VE","w-12":"BaseBanner-module_w-12__YoD9q","w-[18px]":"BaseBanner-module_w---18px--__XTEPc","w-48":"BaseBanner-module_w-48__yKSFa","w-1/4":"BaseBanner-module_w-1--4__kpSQL","min-w-[24px]":"BaseBanner-module_min-w---24px--__8d4U4","max-w-xs":"BaseBanner-module_max-w-xs__3o0ZK","flex-1":"BaseBanner-module_flex-1__9W4gQ","flex-grow":"BaseBanner-module_flex-grow__RWNRg","-translate-x-1/2":"BaseBanner-module_-translate-x-1--2__yg40N","-translate-y-9":"BaseBanner-module_-translate-y-9__rqCjq","translate-x-[22px]":"BaseBanner-module_translate-x---22px--__hA9TX","-translate-y-3":"BaseBanner-module_-translate-y-3__nXByg","transform":"BaseBanner-module_transform__xyxuG","cursor-pointer":"BaseBanner-module_cursor-pointer__TRbnn","cursor-text":"BaseBanner-module_cursor-text__2MNdZ","!cursor-pointer":"BaseBanner-module_--cursor-pointer__YYBqR","cursor-not-allowed":"BaseBanner-module_cursor-not-allowed__ZC557","resize":"BaseBanner-module_resize__YWtIS","appearance-none":"BaseBanner-module_appearance-none__u67l6","flex-row-reverse":"BaseBanner-module_flex-row-reverse__vV-40","flex-col":"BaseBanner-module_flex-col__58Qkw","flex-wrap":"BaseBanner-module_flex-wrap__YDqEB","items-center":"BaseBanner-module_items-center__FEaA8","justify-end":"BaseBanner-module_justify-end__BEkO5","justify-center":"BaseBanner-module_justify-center__9C1WG","gap-1":"BaseBanner-module_gap-1__wUmIB","gap-2":"BaseBanner-module_gap-2__uNE1O","gap-4":"BaseBanner-module_gap-4__nxJEH","gap-3":"BaseBanner-module_gap-3__VksPT","gap-9":"BaseBanner-module_gap-9__Cs7MP","gap-11":"BaseBanner-module_gap-11__foyQi","justify-self-end":"BaseBanner-module_justify-self-end__uvBqD","overflow-hidden":"BaseBanner-module_overflow-hidden__R2E3v","overflow-x-auto":"BaseBanner-module_overflow-x-auto__WLTTY","truncate":"BaseBanner-module_truncate__dOPlv","overflow-ellipsis":"BaseBanner-module_overflow-ellipsis__soiU6","!whitespace-normal":"BaseBanner-module_--whitespace-normal__TGT2a","whitespace-nowrap":"BaseBanner-module_whitespace-nowrap__YXwhh","rounded-3xl":"BaseBanner-module_rounded-3xl__02F7A","rounded":"BaseBanner-module_rounded__uqFwq","rounded-md":"BaseBanner-module_rounded-md__HqVKK","rounded-[3px]":"BaseBanner-module_rounded---3px--__2k1PD","rounded-full":"BaseBanner-module_rounded-full__QtvsX","border":"BaseBanner-module_border__BDwhc","border-2":"BaseBanner-module_border-2__OW-qC","border-none":"BaseBanner-module_border-none__LP24w","border-borders":"BaseBanner-module_border-borders__gyEU7","border-borders-disabled":"BaseBanner-module_border-borders-disabled__ictnF","border-borders-danger":"BaseBanner-module_border-borders-danger__GXq8F","border-blue-500":"BaseBanner-module_border-blue-500__cAYvP","bg-badges-gray":"BaseBanner-module_bg-badges-gray__z4d5j","bg-badges-darkgray":"BaseBanner-module_bg-badges-darkgray__CC79U","bg-badges-red":"BaseBanner-module_bg-badges-red__j46zw","bg-badges-orange":"BaseBanner-module_bg-badges-orange__vZ7x2","bg-badges-blue":"BaseBanner-module_bg-badges-blue__M3-id","bg-badges-green":"BaseBanner-module_bg-badges-green__mkqO1","bg-badges-purple":"BaseBanner-module_bg-badges-purple__0vqw2","bg-badges-green-neon":"BaseBanner-module_bg-badges-green-neon__8jFkh","bg-blue-300":"BaseBanner-module_bg-blue-300__lS8SD","bg-actions-secondary":"BaseBanner-module_bg-actions-secondary__Jpik-","bg-backgrounds-white":"BaseBanner-module_bg-backgrounds-white__TyOpV","bg-backgrounds-overlay":"BaseBanner-module_bg-backgrounds-overlay__gmnme","bg-actions-primary":"BaseBanner-module_bg-actions-primary__y9nRD","bg-actions-primary-disabled":"BaseBanner-module_bg-actions-primary-disabled__6S-uD","bg-feedback-success-content":"BaseBanner-module_bg-feedback-success-content__zc7v7","bg-borders":"BaseBanner-module_bg-borders__Zm-ZZ","bg-borders-disabled":"BaseBanner-module_bg-borders-disabled__vw-mk","bg-blue-500":"BaseBanner-module_bg-blue-500__vO5bY","bg-brand-gray-20":"BaseBanner-module_bg-brand-gray-20__EDuEX","p-2":"BaseBanner-module_p-2__tFhZ-","p-4":"BaseBanner-module_p-4__jkC9M","p-0.5":"BaseBanner-module_p-0--5__BY8S0","p-0":"BaseBanner-module_p-0__7Wokh","p-3":"BaseBanner-module_p-3__M8CzZ","py-0.5":"BaseBanner-module_py-0--5__HIqjL","py-0":"BaseBanner-module_py-0__-guK9","px-1":"BaseBanner-module_px-1__q2Vl9","px-2":"BaseBanner-module_px-2__onxyH","py-1":"BaseBanner-module_py-1__zVsp0","pt-[16px]":"BaseBanner-module_pt---16px--__Dnce4","text-center":"BaseBanner-module_text-center__fcP9G","text-right":"BaseBanner-module_text-right__DfnVK","text-base":"BaseBanner-module_text-base__YhEFZ","text-[20px]":"BaseBanner-module_text---20px--__10yYj","text-[40px]":"BaseBanner-module_text---40px--__pXx9M","text-lg":"BaseBanner-module_text-lg__ZeSSN","text-4xl":"BaseBanner-module_text-4xl__Iah2e","text-xs":"BaseBanner-module_text-xs__4OJtj","text-sm":"BaseBanner-module_text-sm__-Cozy","text-tiny":"BaseBanner-module_text-tiny__PrW4D","text-action-label":"BaseBanner-module_text-action-label__Mu6Jv","text-body-text":"BaseBanner-module_text-body-text__Bs4Kn","text-caption-text":"BaseBanner-module_text-caption-text__NtFy-","text-component-label":"BaseBanner-module_text-component-label__nDH7g","text-page-heading":"BaseBanner-module_text-page-heading__lglhe","text-section-heading":"BaseBanner-module_text-section-heading__kMJ5Z","text-section-subheading":"BaseBanner-module_text-section-subheading__Bh9II","text-subheading":"BaseBanner-module_text-subheading__87kz-","font-bold":"BaseBanner-module_font-bold__NBo88","font-action-label":"BaseBanner-module_font-action-label__crGKr","font-body-text":"BaseBanner-module_font-body-text__lq7Hs","font-caption-text":"BaseBanner-module_font-caption-text__hAwAu","font-component-label":"BaseBanner-module_font-component-label__gM0I6","font-page-heading":"BaseBanner-module_font-page-heading__gUydC","font-section-heading":"BaseBanner-module_font-section-heading__c4dwE","font-section-subheading":"BaseBanner-module_font-section-subheading__vK3-a","font-subheading":"BaseBanner-module_font-subheading__8j-MW","uppercase":"BaseBanner-module_uppercase__FVRax","italic":"BaseBanner-module_italic__enwi8","text-badges-red":"BaseBanner-module_text-badges-red__nHj82","text-badges-orange":"BaseBanner-module_text-badges-orange__AtQ3g","text-badges-blue":"BaseBanner-module_text-badges-blue__Rjtr-","text-badges-green":"BaseBanner-module_text-badges-green__t9YNP","text-badges-darkgray":"BaseBanner-module_text-badges-darkgray__8VmZ-","text-badges-purple":"BaseBanner-module_text-badges-purple__wK1n2","text-badges-green-neon":"BaseBanner-module_text-badges-green-neon__n1nJJ","text-white":"BaseBanner-module_text-white__zxLyC","text-neutrals-primary-light":"BaseBanner-module_text-neutrals-primary-light__pwiOn","text-actions-primary":"BaseBanner-module_text-actions-primary__TS06X","!text-actions-primary":"BaseBanner-module_--text-actions-primary__hq8J8","text-feedback-error-content":"BaseBanner-module_text-feedback-error-content__Zbwqv","text-neutrals-tertiary":"BaseBanner-module_text-neutrals-tertiary__TZ5cd","text-neutrals-disabled":"BaseBanner-module_text-neutrals-disabled__DvMW-","!text-actions-primary-disabled":"BaseBanner-module_--text-actions-primary-disabled__klLvZ","text-blue-500":"BaseBanner-module_text-blue-500__7DOoG","text-gray-100":"BaseBanner-module_text-gray-100__5qoQi","text-neutrals-primary":"BaseBanner-module_text-neutrals-primary__VXF1K","text-neutrals-secondary":"BaseBanner-module_text-neutrals-secondary__8avtk","underline":"BaseBanner-module_underline__WI-z9","opacity-0":"BaseBanner-module_opacity-0__ID7Qa","shadow":"BaseBanner-module_shadow__Mt17N","shadow-5":"BaseBanner-module_shadow-5__KsU8i","outline-none":"BaseBanner-module_outline-none__oTfbm","filter":"BaseBanner-module_filter__AlnAQ","!filter":"BaseBanner-module_--filter__bbceF","transition-transform":"BaseBanner-module_transition-transform__0Xest","transition":"BaseBanner-module_transition__dQqlZ","duration-300":"BaseBanner-module_duration-300__4DQnX","info":"BaseBanner-module_info__FbizJ","warning":"BaseBanner-module_warning__f5Lp2","error":"BaseBanner-module_error__TNsS6","success":"BaseBanner-module_success__nXJlH","icon-info":"BaseBanner-module_icon-info__FNDhb","icon-warning":"BaseBanner-module_icon-warning__NHSpA","icon-error":"BaseBanner-module_icon-error__ClJ0q","icon-success":"BaseBanner-module_icon-success__hzwYQ","focus-within:outline-none":"BaseBanner-module_focus-within--outline-none__bXiCB","hover:bg-actions-secondary-hover":"BaseBanner-module_hover--bg-actions-secondary-hover__GAMW0","hover:shadow-elevation-10":"BaseBanner-module_hover--shadow-elevation-10__JnXqj","focus:border-actions-primary":"BaseBanner-module_focus--border-actions-primary__Jf7dJ","focus:border-actions-danger-active":"BaseBanner-module_focus--border-actions-danger-active__6CGBk","focus-visible:rounded-md":"BaseBanner-module_focus-visible--rounded-md__obM-8","focus-visible:border-2":"BaseBanner-module_focus-visible--border-2__RtsET","focus-visible:border-actions-primary-active":"BaseBanner-module_focus-visible--border-actions-primary-active__Qta92","focus-visible:bg-actions-secondary-active":"BaseBanner-module_focus-visible--bg-actions-secondary-active__ShmZw","group":"BaseBanner-module_group__7lPWf","group-hover:block":"BaseBanner-module_group-hover--block__gygWV","group-hover:translate-y-0":"BaseBanner-module_group-hover--translate-y-0__FbnSj","group-hover:transform":"BaseBanner-module_group-hover--transform__QE1mx","group-hover:opacity-100":"BaseBanner-module_group-hover--opacity-100__NJBLU","group-focus:translate-y-0":"BaseBanner-module_group-focus--translate-y-0__NVjKW","group-focus:transform":"BaseBanner-module_group-focus--transform__PcuiE","group-focus:opacity-100":"BaseBanner-module_group-focus--opacity-100__00ugI"};

var BaseBanner = function (_a) {
    var className = _a.className, _b = _a.dataTestId, dataTestId = _b === void 0 ? 'base-banner-component' : _b, _c = _a.type, type = _c === void 0 ? 'info' : _c, children = _a.children, _d = _a.large, large = _d === void 0 ? false : _d, icon = _a.icon, _e = _a.closeButton, closeButton = _e === void 0 ? false : _e, onClose = _a.onClose, linkLabel = _a.linkLabel, onLinkClick = _a.onLinkClick;
    return (jsxRuntime.jsx("div", tslib.__assign({ className: cx("relative w-full flex items-center border ".concat(styles[type]), className), "data-testid": dataTestId }, { children: jsxRuntime.jsxs("div", tslib.__assign({ className: cx('w-full flex items-center p-2 gap-2', {
                'p-4 gap-4': large,
            }) }, { children: [icon && (jsxRuntime.jsx("div", tslib.__assign({ className: "w-6 h-6 rounded flex justify-center items-center ".concat(styles['icon-' + type]) }, { children: jsxRuntime.jsx(Icon, { name: icon, className: cx('text-neutrals-primary-light rounded text-[20px]', {
                            'text-[40px]': large,
                        }) }) }))), jsxRuntime.jsx("div", tslib.__assign({ className: cx('flex-grow overflow-hidden truncate whitespace-nowrap', {
                        '!whitespace-normal': large,
                    }) }, { children: children })), jsxRuntime.jsxs("div", tslib.__assign({ className: "flex justify-self-end" }, { children: [linkLabel && (jsxRuntime.jsx("button", tslib.__assign({ className: "flex items-center", onClick: onLinkClick }, { children: jsxRuntime.jsx(ComponentLabel, tslib.__assign({ className: "text-actions-primary cursor-pointer whitespace-nowrap" }, { children: linkLabel })) }))), closeButton && (jsxRuntime.jsx("button", tslib.__assign({ className: "w-10 h-10 flex justify-center items-center focus-within:outline-none", onClick: onClose, "data-testid": "close-button" }, { children: jsxRuntime.jsx(Icon, { name: "close_big" }) })))] }))] })) })));
};

var NetworkBannerStatus = {
    offline: {
        bannerType: 'warning',
        icon: 'circle_check',
        label: 'No internet. Showing cached data from last online session',
        linkLabel: 'Learn more',
    },
    syncing: {
        bannerType: 'info',
        icon: 'cloud_check',
        label: 'Internet connection restored. Syncing offline changes',
    },
    synced: {
        bannerType: 'success',
        icon: 'circle_check',
        label: 'Offline changes synced!',
    },
    syncFailed: {
        bannerType: 'error',
        icon: 'error',
        label: 'Failed to sync offline changes',
        linkLabel: 'Retry',
    },
    offlineForm: {
        bannerType: 'info',
        icon: 'info_circle',
        label: 'You are currently offline. Your task actions will be synced as soon as you are back online. Please do not refresh or close the browser',
    },
    offlineFormSubmitted: {
        bannerType: 'success',
        icon: 'circle_check',
        label: 'Task submitted offline',
    },
    offlineAttachmentsSubmitted: {
        bannerType: 'success',
        icon: 'circle_check',
        label: 'Attachment submitted while in offline, the action will be synced once we comeback online',
    },
    offlineAttachmentsDeleted: {
        bannerType: 'success',
        icon: 'circle_check',
        label: 'Attachment/s deleted while in offline',
    },
    maintenanceScheduled: {
        bannerType: 'warning',
        icon: 'info_circle',
        label: 'Maintenance scheduled on {maintenanceDate} at {maintenanceStartTime} until {maintenanceEndTime}',
    },
    maintenanceInProgress: {
        bannerType: 'warning',
        icon: 'info_circle',
        label: 'Maintenance in progress',
    },
    maintenanceCompleted: {
        bannerType: 'success',
        icon: 'circle_check',
        label: 'Maintenance completed',
    },
    maintenanceFailed: {
        bannerType: 'error',
        icon: 'error',
        label: 'Maintenance failed',
    },
    releaseScheduled: {
        bannerType: 'info',
        icon: 'info_circle',
        label: "Release scheduled on {releaseDate} at {releaseTime} (v{version})",
        linkLabel: 'Release notes',
        closeButton: true,
    },
};

var NetworkStatusBanner = function (_a) {
    var variant = _a.variant, icon = _a.icon, label = _a.label, linkLabel = _a.linkLabel, metadata = _a.metadata, closeButton = _a.closeButton, onLinkClick = _a.onLinkClick, onClose = _a.onClose;
    var _b = React.useState(true), isOpen = _b[0], setOpen = _b[1];
    if (!isOpen) {
        return null;
    }
    var handleClose = function () {
        setOpen(false);
        onClose === null || onClose === void 0 ? void 0 : onClose();
    };
    var handleLinkClick = function () {
        onLinkClick === null || onLinkClick === void 0 ? void 0 : onLinkClick();
    };
    var _c = NetworkBannerStatus, _d = variant, _e = _c[_d], bannerType = _e.bannerType, defaultIcon = _e.icon, defaultLabel = _e.label, defaultLinkLabel = _e.linkLabel, large = _e.large, defaultCloseButton = _e.closeButton;
    var replaceMetadataKeys = function (label) {
        if (!metadata) {
            return label;
        }
        var replacedLabel = label;
        Object.keys(metadata !== null && metadata !== void 0 ? metadata : {}).forEach(function (key) {
            replacedLabel = replacedLabel.replace("{".concat(key, "}"), metadata[key] || '');
        });
        return replacedLabel;
    };
    var finalLabel = label
        ? replaceMetadataKeys(label)
        : replaceMetadataKeys(defaultLabel);
    return (jsxRuntime.jsx(BaseBanner, tslib.__assign({ type: bannerType, icon: icon ? icon : defaultIcon, large: large, closeButton: closeButton !== null && closeButton !== void 0 ? closeButton : defaultCloseButton, onClose: handleClose, onLinkClick: handleLinkClick, linkLabel: linkLabel ? linkLabel : defaultLinkLabel }, { children: finalLabel })));
};

var Chip = function (_a) {
    var id = _a.id, label = _a.label, _b = _a.clickable, clickable = _b === void 0 ? false : _b, _c = _a.deletable, deletable = _c === void 0 ? false : _c, onClick = _a.onClick, onClose = _a.onClose;
    var chipId = id + '';
    var handleClick = function () {
        if (clickable && onClick) {
            onClick({ id: chipId });
        }
    };
    var handleClose = function (e) {
        e.stopPropagation();
        if (onClose) {
            onClose({ id: chipId });
        }
    };
    return (jsxRuntime.jsxs("div", tslib.__assign({ className: "w-max flex items-center gap-1 bg-actions-secondary border border-borders z-10 px-1 py-0.5 rounded-md hover:bg-actions-secondary-hover" }, { children: [jsxRuntime.jsx("button", tslib.__assign({ className: cx('outline-none cursor-text focus-visible:border-2 focus-visible:border-actions-primary-active focus-visible:bg-actions-secondary-active focus-visible:rounded-md', {
                    '!cursor-pointer': clickable,
                }), onClick: handleClick, "data-testid": "button-chip" }, { children: jsxRuntime.jsx(BodyText, tslib.__assign({ className: cx('pointer-events-none', {
                        'cursor-pointer !text-actions-primary': clickable,
                    }) }, { children: label })) })), deletable && (jsxRuntime.jsx("button", tslib.__assign({ className: "flex items-center appearance-none outline-none z-20 focus-visible:border-2 focus-visible:border-actions-primary-active hover:bg-actions-secondary-hover rounded-md p-0.5 cursor-pointer", onClick: handleClose, "data-testid": "button-delete" }, { children: jsxRuntime.jsx(Icon, { name: "off_outline_close", className: "text-[20px]" }) })))] })));
};

var ChipGroup = function (_a) {
    var className = _a.className, _b = _a.orientation, orientation = _b === void 0 ? 'horizontal' : _b, children = _a.children;
    return (jsxRuntime.jsx("div", tslib.__assign({ className: cx('flex flex-wrap gap-2', className, {
            'flex-col': orientation === 'vertical',
        }), "data-testid": "chip-group" }, { children: children })));
};

exports.ActionLabel = ActionLabel;
exports.Badge = Badge$1;
exports.Banner = Banner;
exports.BaseBanner = BaseBanner;
exports.BodyText = BodyText;
exports.CaptionText = CaptionText;
exports.Chip = Chip;
exports.ChipGroup = ChipGroup;
exports.ComponentLabel = ComponentLabel;
exports.Icon = Icon;
exports.NetworkStatusBanner = NetworkStatusBanner;
exports.PageHeading = PageHeading;
exports.Portal = Portal;
exports.SectionHeading = SectionHeading;
exports.SectionSubheading = SectionSubheading;
exports.Slider = Slider;
exports.Subheading = Subheading;
exports.UrbintBadge = Badge;
